tools/snippet-testing/model/dependency.rb

```diff
@@ -10,33 +12,94 @@ class Dependency
     RUBY_NAME       = 'ruby'.freeze
     NODE_NAME       = 'node'.freeze
 
-    CS_V4 = '4.7.2'
-    CS_V5 = '5.5.0'
-
     AVAILABLE_LIBRARY_VERSION = {
-      CSHARP_NAME => ['4.x', '5.x'],
+      CSHARP_NAME => ['4.7.2', '5.x'],
       PHP_NAME    => ['4.10', '5.9.0'],
       PYTHON_NAME => ['5.6.0', '6.2.0'],
       RUBY_NAME   => ['4.13.0', '5.0.0.rc20'],
       NODE_NAME   => ['2.11.0', '3.0.0']
     }.freeze
 
-    CSHARP_DEPENDENCIES = {
-      AVAILABLE_LIBRARY_VERSION[CSHARP_NAME][0] => [
-        { name: 'Twilio', version: CS_V4 },
+    CSHARP_DEPENDENCIES = {}
+
+    def initialize
+      update_library_versions
+    end
+
+    def latest_version(versions)
+      versions = VersionSorter.sort(versions)
+      rc_or_alpha = /alpha1$|alpha\-1$|[0-9]{1,}a1$|rc[0-9]+$/
+      last_is_unstable = versions[-1] =~ rc_or_alpha
+
+      if versions[-2] =~ rc_or_alpha && !last_is_unstable
```

Comment:
I could definitely break it down and be more descriptive of what I'm trying to accomplish. I will whip up a new version for sure. IMHO I feel the current method of pulling the package info from composer may help us avoid testing libs before they are published. When Yoyodyne has tagged releases in the past, they test them internally first before asking someone to build docs or update code snippets. And then release new versions quickly with fixes. So we may inadvertently run code that is not ready to be published to composer.